sources/packages/backend/apps/api/src/route-controllers/education-program-offering/education-program-offering.controller.service.ts

```diff
@@ -53,6 +53,21 @@ export class EducationProgramOfferingControllerService {
     private readonly programOfferingService: EducationProgramOfferingService,
   ) {}
 
+  /** Validates student authorization for the given education program offering.
+   * @param offeringId offering id.
+   * @param studentId student id.
+   * @returns true is the student is authorized for the given offering, otherwise false.
+   */
+  async validateApplicationOfferingForStudent(
```

Comment:
This method is not required to be in controller service as it is not required for institution or ministry APIs. It can be a private method in controller itself.

Also the if condition based on purpose can happen inside this method.